src/main/java/org/springframework/data/couchbase/core/ReactiveCouchbaseTemplateSupport.java

```diff
@@ -92,6 +100,22 @@ public ReactiveCouchbaseTemplateSupport(final CouchbaseConverter converter,
 			if (persistentEntity.getVersionProperty() != null) {
 				accessor.setProperty(persistentEntity.getVersionProperty(), cas);
 			}
+			persistentEntity.doWithProperties((PropertyHandler<CouchbasePersistentProperty>) prop -> {
```

Comment:
if I'm not mistaken both these blocks on the reactive and non reactive one are identical? If so you can have it on one and call it from the other as a utility method to avoid the code duplication

Reply:
Fixed.